app/models/pawn.rb

```diff
@@ -7,4 +7,19 @@ def get_image
       '&#9823;'
     end
   end
+
+  def valid_move?(x,y)
+    if super(x,y)
+      movement_direction = white? ? 1 : -1
+      # standard movement
+      if x == x_coord && square_is_vacant?(x,y)
```

Comment:
Let's try to refactor it out a little bit, maybe abstracting into separate methods like we discussed during the call? 

Ideally we would end with the `valid_move?` method being a container/summary of what's happening.. Sth like: 

```
def valid_move?(x,y)
  super(x,y) && (valid_one_forward_move?(x,y) || valid_two_forward_move?(x,y) || valid_diagonal_move?(x,y))
end
```

I also like the idea of using `movement_direction` variable, so maybe you can pass it to the corresponding methods...

Hope that makes sense

Reply:
Yes that makes total sense! It also makes it more extensible were we to add en passant capture later. Check out my recent changes @ilyakrasnov !